be/src/io/cache/file_cache_manager.cpp

```diff
@@ -18,15 +18,39 @@
 #include "io/cache/file_cache_manager.h"
 
 #include "gutil/strings/util.h"
+#include "io/cache/dummy_file_cache.h"
 #include "io/cache/sub_file_cache.h"
 #include "io/cache/whole_file_cache.h"
 #include "io/fs/local_file_system.h"
+#include "olap/storage_engine.h"
 #include "util/file_utils.h"
 #include "util/string_util.h"
 
 namespace doris {
 namespace io {
 
+void GCContextPerDisk::init(const std::string& path, int64_t max_size) {
+    _disk_path = path;
+    _conf_max_size = max_size;
+    _used_size = 0;
+}
+bool GCContextPerDisk::try_add_file_cache(FileCachePtr cache, int64_t file_size) {
+    if (cache->cache_dir().string().substr(0, _disk_path.size()) == _disk_path) {
```

Comment:
May be we should determine whether the cache path exists in the _lru_queue? 
For example, When there is a cache file in the disk but not in the memory, the cache will be add to the _lru_queue. But when the cache needs to be used to create the cache in the memory, the cache pointer will also be added to the _lru_queue in the cleanup task. The cache last match time in the lru queue will be earlier, which may cause the cache to be cleared in advance?

Reply:
same as above